src/Bonsai.Zyre/ZyreNode.cs

```diff
@@ -0,0 +1,277 @@
+        /// </summary>
+        /// <returns>
+        /// An observable sequence of <see cref="ZyreEvent"/> events received from other members of the group.
+        /// </returns>
+        public override IObservable<ZyreEvent> Generate()
+        {
+            return Generate(null);
+        }
+
+        /// <summary>
+        /// Creates a Zyre node that joins a network group, listens for events from other group peers,
+        /// and transmitss an observable sequence of messages back to the group.
+        /// </summary>
+        /// <param name="source">
+        /// The sequence of messages to transmit.
+        /// </param>
+        /// <returns>
+        /// An observable sequence of <see cref="ZyreEvent"/> events received from other members of the group.
+        /// </returns>
+        public IObservable<ZyreEvent> Generate(IObservable<ZyreMessage> source)
+        {
+            return Observable.Create<ZyreEvent>(observer =>
+            {
+                NetMQZyre.Zyre zyre = new NetMQZyre.Zyre(Name);
+                zyre.Join(Group);
+                zyre.SetInterface(Interface);
+                zyre.Start();
+
+                if (source != null)
+                {
+                    var message = source.Do(m =>
+                    {
+                        m.Send(zyre);
+                    }).Subscribe();
+                }
+
+                // A peer has joined the network.
+                zyre.EnterEvent += (sender, e) =>
+                {
+                    observer.OnNext(new ZyreEvent
+                    {
+                        EventType = nameof(zyre.EnterEvent),
+                        FromNode = e.SenderName,
+                        FromNodeUid = e.SenderUuid,
+                        Content = new NetMQMessage(new List<NetMQFrame> { NetMQFrame.Empty })
+                    });
+                };
+
+                // A peer is being evasive (quiet).
+                zyre.EvasiveEvent += (sender, e) =>
+                {
+                    observer.OnNext(new ZyreEvent
+                    {
+                        EventType = nameof(zyre.EvasiveEvent),
+                        FromNode = e.SenderName,
+                        FromNodeUid = e.SenderUuid,
+                        Content = new NetMQMessage(new List<NetMQFrame> { NetMQFrame.Empty })
+                    });
+                };
+
+                // A peer has left the network.
+                zyre.ExitEvent += (sender, e) =>
+                {
+                    observer.OnNext(new ZyreEvent
+                    {
+                        EventType = nameof(zyre.ExitEvent),
+                        FromNode = e.SenderName,
+                        FromNodeUid = e.SenderUuid,
+                        Content = new NetMQMessage(new List<NetMQFrame> { NetMQFrame.Empty })
+                    });
+                };
+
+                // A peer has joined a specific group.
```

Comment:
Do these events carry group information? This relates to my earlier question about whether Zyre nodes can join multiple groups.